### PR TITLE
Relative URL resolution improved and other code improvements.

### PR DESCRIPTION
**Pull request by farkaskid**
Addresses #16

### CODE REVIEW

File: crawler/collector.go [hunk 1]
```diff
@@ -24,26 +25,21 @@ func (collector *URLCollector) sync(f func()) {
 	collector.Unlock()
 }
 
-func (collector *URLCollector) Visited(m map[uint64]bool, s string) (visited bool) {
+func (collector *URLCollector) visited(m map[uint64]bool, s string) (visited bool) {
```
Comment (mahesh1996) [inline crawler/collector.go:28]: Does method starting with small caps char private?

**Review by mahesh1996 — APPROVED** [review]

File: crawler/collector.go (hunk 1 above)
Comment (farkaskid) [reply crawler/collector.go:28]: Yeah.

Comment (mahesh1996) [reply crawler/collector.go:28]: Was it previously being used as publically and now not being used as publically?

Comment (farkaskid) [reply crawler/collector.go:28]: There is/was no public usage. That's why I lowercased it.

**Pull request merged**